msdk-id-sirius/src/main/java/io/github/msdk/id/sirius/FingerIdWebMethod.java

```diff
@@ -0,0 +1,646 @@
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.TimeoutException;
+
+import javax.annotation.Nullable;
+import org.apache.http.HttpEntity;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+
+import org.openscience.cdk.DefaultChemObjectBuilder;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IMolecularFormula;
+import org.openscience.cdk.smiles.SmilesParser;
+import org.openscience.cdk.tools.manipulator.MolecularFormulaManipulator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>Class FingerIdWebMethod</p>
+ * This class wraps the FingerId API provided by boecker-labs
+ * Uses the results of SiriusIdentificationMethod and returns exteneded IonAnnotations
+ */
+public class FingerIdWebMethod implements MSDKMethod<List<IonAnnotation>> {
+
+  private final static SmilesParser smp = new SmilesParser(DefaultChemObjectBuilder.getInstance());
+  private final static CloseableHttpClient client = HttpClients.createSystem(); // Threadsafe
+  private final static Logger logger = LoggerFactory.getLogger(FingerIdWebMethod.class);
+  private final static BasicNameValuePair UID = new BasicNameValuePair("uid", SystemInformation.generateSystemKey());
+  private final static String FINGERID_SOURCE = "https://www.csi-fingerid.uni-jena.de";
+  /* TODO: This field must be altered whenever boecker-labs updates its API!!! */
+  private final static String FINGERID_VERSION = "1.1.2";
+  private final static SearchStructureByFormula searchDB = new RESTDatabase(BioFilter.ALL);
+  private final static Gson gson = new Gson();
+
+  private final Ms2Experiment experiment;
+  private final SiriusIonAnnotation ionAnnotation;
+  private final PredictionPerformance[] perf;
+  private final MaskedFingerprintVersion version;
+  private final Fingerblast blaster;
+  private final int candidatesAmount;
+  private List<IonAnnotation> newAnnotations;
+  private int finishedItems;
+  private boolean cancelled;
+
+
+  /**
+   * <p>Constructor for a FingerIdWebMethod</p>
+   * @param experiment - Ms2Experiment that was used to get ionAnnotation
+   * @param ionAnnotation - SiriusIonAnnotation returned from SiriusIdentificationMethod with FTree field specified
+   * @param candidatesAmount - amount of candidates to be returned from this method
+   * @throws MSDKException if any
+   */
+  public FingerIdWebMethod(Ms2Experiment experiment, SiriusIonAnnotation ionAnnotation, int candidatesAmount) throws MSDKException {
```

Comment:
Please add @Nullable or @Nonnull annotations to all parameters of public methods. Also, it is better to use Integer instead of int for the parameters.